### ISSUE TITLE
create_sql_agent has incorrect behaviour when it queries agaisnt Google BigQuery

### ISSUE THREAD

**Issue by moonbug-car** [langchain-ai/langchain#10225]
### System Info

OS: Ubuntu 22.04.3 LTS

SQLAlchemy==1.4.49
langchain==0.0.281

### Information

- [X] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [X] Agents / Agent Executors
- [X] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

```
from langchain.agents import create_sql_agent
from langchain.agents.agent_toolkits import SQLDatabaseToolkit
from langchain.llms import VertexAI
from langchain.sql_database import SQLDatabase

project = "AAAA"
dataset = "HHHH"
sqlalchemy_url = f'bigquery://{project}/{dataset}'

db = SQLDatabase.from_uri(sqlalchemy_url)

llm = VertexAI(
    model_name="text-bison@001",
    max_output_tokens=256,
    temperature=0.1,
    top_p=0.8,
    top_k=40,
    verbose=True,
)
toolkit = SQLDatabaseToolkit(db=db, llm=llm)

agent_executor = create_sql_agent(
    llm=llm,
    toolkit=toolkit,
    verbose=True, 
    max_execution_time=60,
)

question = "What is the total <numbers> in 2023?" #replace number to the target metric

response = agent_executor.run(question)

```
The agent will return "(Background on this error at: https://sqlalche.me/e/14/4xp6) Error: (google.cloud.bigquery.dbapi.exceptions.DatabaseError) 400 Syntax error: Expected end of input but got identifier "Thought" at [4:1]"
And when I check the project history on GCP, I found the agent included the thought and action inside the query, so the executed query was like that:
```
SELECT SUM(AAA) AS total_AAA, SUM(BBB) AS total_BBB
FROM YOU_CANT_SEE_ME
WHERE Year = 2023
Thought: I should check the query before executing it.
Action: sql_db_query_checker
Action Input: 
SELECT SUM(AAA) AS total_AAA, SUM(BBB) AS total_BBB
FROM YOU_CANT_SEE_ME
WHERE Year = 2023
```

It was stunning. Or was there anything I did wrong?

### Expected behavior

It should phrase a correct query, execute it and return a legit result. It works fine when I do not specify the year.

**Comment by moonbug-car** [comment]
@dosu-bot 
Tried. No luck. =(

**Comment by moonbug-car** [comment]
Waiting for human response.

**Comment by tiny-tinker** [comment]
I was looking for examples of BigQuery and this create_sql_agent function and found this post. I successfully followed the instructions on [this page](https://www.datascienceengineer.com/blog/post-chat-with-bigquery) and did not get any such errors. I tried adding in the `<` and `>`, but the queries executed successfully. 

I am using `text-bison@latest` but that really shouldn't matter. 

Also, this is a great post for why we're not terribly worried about losing our jobs yet :D
